### PR TITLE
L2-597: Detach Canister

### PR DESCRIPTION
**Pull request by lmuntaner**
# Motivation

User can detach a canister.

# Changes

* Add a DetachCanisterButton in CanisterDetails route.
* New detachCanister method in NNSDappCanister.
* New canister api detachCanister.
* New canister service detachCanister.

# Tests

* Test user flow to detach a canister.
* Test for NNSDappCanister method detachCanister and attachCanister (I forgot before).
* Test api function detachCanister.
* Test canister service detachCanister.

### CODE REVIEW

File: frontend/svelte/src/lib/services/canisters.services.ts [hunk 1]
```diff
@@ -127,6 +128,23 @@ export const attachCanister = async (
...
+export const detachCanister = async (
+  canisterId: Principal
+): Promise<{ success: boolean }> => {
+  try {
+    const identity = await getIdentity();
+    await detachCanisterApi({
+      identity,
+      canisterId,
+    });
+    await listCanisters({ clearBeforeQuery: false });
```
Comment (peterpeterparker) [inline frontend/svelte/src/lib/services/canisters.services.ts:140]: edge case: what happens if detach is success and list canisters fails?

currently user would remain on detail page of a canister that has been detached

File: frontend/svelte/src/lib/components/canister_details/DetachCanisterButton.svelte [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    const { success } = await detachCanister(canisterId);
+    stopBusy("detach-canister");
+    if (success) {
+      toastsStore.show({
+        level: "success",
+        labelKey: "canister_detail.detach_success",
+      });
+      routeStore.replace({ path: AppPath.Canisters });
+    }
+    close();
```
[the remark this answers is not in the document]
Comment (lmuntaner) [reply frontend/svelte/src/lib/components/canister_details/DetachCanisterButton.svelte:30]: good point, yes I can first close and then the other part.

File: frontend/svelte/src/lib/services/canisters.services.ts (hunk 1 above)
Comment (lmuntaner) [reply frontend/svelte/src/lib/services/canisters.services.ts:142]: done

File: frontend/svelte/src/lib/components/canister_details/DetachCanisterButton.svelte (hunk 2 above)
[the remark this answers is not in the document]
Comment (lmuntaner) [reply frontend/svelte/src/lib/components/canister_details/DetachCanisterButton.svelte]: done

**Comment by lmuntaner** [comment]
@peterpeterparker ready for next review

**Review by peterpeterparker — APPROVED** [review]
LGTM, thx

**Pull request merged**